samples/dependadotnet/Program.cs

```diff
@@ -0,0 +1,84 @@
+﻿using System;
+using System.IO;
+using System.Text.RegularExpressions;
+using static System.Console;
+
+if (args is { Length: 0 } || args[0] is not string path)
+{
+    WriteLine("Must specify a repo root directory as input");
+    return;
+}
+
+// yaml top-matter
+
+string topMatter =
+@"# generated by dependadotnet
+# https://github.com/dotnet/core/tree/master/samples/dependadotnet
+version: 2
+updates:";
+
+WriteLine(topMatter);
+
+/* Generate the following pattern for each project file:
+
+  Note: Wednesday was chosen for quick response to .NET patch Tuesday updates
+
+- package-ecosystem: ""nuget""
+  directory: ""/"" #projectfilename
+  schedule:
+      interval: ""weekly""
+      day: ""wednesday""
+  open-pull-requests-limit: 5
+*/
+
+string regex = @"PackageReference.*Version=""[0-9]";
+
+foreach (string file in Directory.EnumerateFiles(path,"*.*",SearchOption.AllDirectories))
+{
+    if (!IsProject(file))
+    {
+        continue;
+    }
+
+
+    string filename = Path.GetFileName(file);
+    string? parentDir = Path.GetDirectoryName(file);
+    string relativeDir = parentDir?.Substring(path.Length).Replace('\\','/') ?? "/";
```

Comment:
Consider using the `SeparatorChar` and `AltSeparatorChar`. This should work on both Windows and unix:

```suggestion
    string relativeDir = parentDir?.Substring(path.Length).Replace(Path.DirectorySeparatorChar,Path.AltDirectorySeparatorChar) ?? Path.AltDirectorySeparatorChar.ToString();
```